src/Analyser/ContainerConstantResolver.php

```diff
@@ -0,0 +1,24 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Analyser;
+
+use PHPStan\DependencyInjection\Container;
+use PHPStan\Reflection\ReflectionProvider;
+
+class ContainerConstantResolver extends ConstantResolver
```

Comment:
I guess that's what `ReflectionProviderProvider` is for :) Try that first, thanks.

Reply:
For reference: phpstan/phpstan#1740 was the reason for injecting the container instead of injecting the `dynamicConstantNames`

Reply:
In that case - everything is the same as I described above, but instead of `arguments` with dynamicConstantNames, add a ConstantResolverFactory that creates ConstantResolver with the array from the Container.

I can adjust it myself if you feel lost a bit, I'm fluent in the nette/di depths 😊

Reply:
I'll get on it next thing tomorrow 👍